source/dem/particle_particle_contact_force.cc

```diff
@@ -11,88 +11,10 @@ template <
-        properties.restitution_coefficient_particle.at(i);
-      const double friction_coefficient_i =
-        properties.friction_coefficient_particle.at(i);
-      const double rolling_friction_coefficient_i =
-        properties.rolling_friction_coefficient_particle.at(i);
-      const double surface_energy_i = properties.surface_energy_particle.at(i);
-
-      for (unsigned int j = 0; j < n_particle_types; ++j)
-        {
-          const unsigned int k = i * n_particle_types + j;
-
-          const double youngs_modulus_j =
-            properties.youngs_modulus_particle.at(j);
-          const double poisson_ratio_j =
-            properties.poisson_ratio_particle.at(j);
-          const double restitution_coefficient_j =
-            properties.restitution_coefficient_particle.at(j);
-          const double friction_coefficient_j =
-            properties.friction_coefficient_particle.at(j);
-          const double rolling_friction_coefficient_j =
-            properties.rolling_friction_coefficient_particle.at(j);
-          const double surface_energy_j =
-            properties.surface_energy_particle.at(j);
-
-          this->effective_youngs_modulus[k] =
-            (youngs_modulus_i * youngs_modulus_j) /
-            ((youngs_modulus_j * (1.0 - poisson_ratio_i * poisson_ratio_i)) +
-             (youngs_modulus_i * (1.0 - poisson_ratio_j * poisson_ratio_j)) +
-             DBL_MIN);
-
-          this->effective_shear_modulus[k] =
-            (youngs_modulus_i * youngs_modulus_j) /
-            (2.0 * ((youngs_modulus_j * (2.0 - poisson_ratio_i) *
-                     (1.0 + poisson_ratio_i)) +
-                    (youngs_modulus_i * (2.0 - poisson_ratio_j) *
-                     (1.0 + poisson_ratio_j))) +
-             DBL_MIN);
-
-          this->effective_coefficient_of_restitution[k] =
-            harmonic_mean(restitution_coefficient_i, restitution_coefficient_j);
-
-          this->effective_coefficient_of_friction[k] =
-            harmonic_mean(friction_coefficient_i, friction_coefficient_j);
-
-          this->effective_coefficient_of_rolling_friction[k] =
-            harmonic_mean(rolling_friction_coefficient_i,
-                          rolling_friction_coefficient_j);
-
-          this->effective_surface_energy[k] =
-            surface_energy_i + surface_energy_j -
-            std::pow(std::sqrt(surface_energy_i) - std::sqrt(surface_energy_j),
-                     2);
-
-          double restitution_coefficient_particle_log =
-            std::log(this->effective_coefficient_of_restitution[k]);
-
-          this->model_parameter_beta[k] =
-            restitution_coefficient_particle_log /
-            sqrt(restitution_coefficient_particle_log *
-                   restitution_coefficient_particle_log +
-                 9.8696);
-        }
-    }
-}
+  : force_calculation_threshold_distance(
+      this->template set_force_calculation_threshold_distance<contact_model>(
+        dem_parameters))
```

Comment:
I really do not like this. Making this a member of the class will mean that every access to this variable will require a this-> pointer dereferencing. Why not calculate it locally at the beggining of the force calculation and store it into a const double? It's not a good idea to keep calculation variable as class members.

Reply:
so at the beginning of calculate_particle_particle_contact_force, doing a if constexpr ? 

Sure this works. It's a tiny calculation. It will be done at every time step, but it is done once.

Reply:
Just confirm if I understood what you meant @blaisb

Reply:
Mhm, I don't really understand Bruno, why this is better to calculate this for every contacts even if this value really need to be calculated only once no matter the particle type (it takes the min/max).

Reply:
Because everytime you are accessing a member variable you will be going through the this-> pointer indirection.
The best thing to do is just calculate it at the beginning of the function and not before every contact. Why store it as a member variable that you need to carry everywhere if you can just recalculate it and keep it locally before you do the force calculation, that's a bit my takeway. The same thing could have been done for the property table. My goal is to keep the classes as lean as possible and keep the information as close as possible to the actual calculation.

Reply:
But before your tables, it was calculated everytime and it was proven to be faster that was, no?

Reply:
Nah we always made it with tables. Initially it's because we were calculating it IN the force model. Calculating it once per iteration would have no consequences.